src/name_core.jl

```diff
@@ -241,22 +241,20 @@ Given a tuple of dimension names, and either a collection of dimensions,
 or a single dimension, expressed as a number,
 Returns the dimension names with those dimensions dropped.
 """
-function remaining_dimnames_after_dropping(dimnames::Tuple, dropped_dim::Integer)
+function remaining_dimnames_after_dropping(dimnames::Tuple, dropped_dim::Int)
     # 0 allocations. See `@btime remaining_dimnames_after_dropping((:a,:b,:c,:d,:e), 4)`
-    return remaining_dimnames_after_dropping(dimnames, (dropped_dim,))
+    return _remaining_dimnames_after_dropping(dimnames, Tuple{dropped_dim})
 end
 
-function remaining_dimnames_after_dropping(dimnames::Tuple, dropped_dims)
-    # 0-Allocations see: `@btime remaining_dimnames_after_dropping((:a,:b,:c,:d,:e), (1,2,))
-
-    anti_names = identity_namedtuple(map(x->dimnames[x], dropped_dims))
-    full_names = identity_namedtuple(dimnames)
+function remaining_dimnames_after_dropping(dimnames::NTuple{N, Symbol}, dropped_dims::Tuple{Vararg{Int}}) where N
+    # 0-Allocations see:
+    # `@code_typed (()->remaining_dimnames_after_dropping((:a,:b,:c,:d,:e), (1,3)))()`
+    return _remaining_dimnames_after_dropping(dimnames, Tuple{dropped_dims...})
+end
 
-    # Now we construct a new named tuple, with all the names we want to remove at the start
-    combined_names = merge(anti_names, full_names)
-    n_skip = length(anti_names)
-    ret = ntuple(length(full_names) - n_skip) do ii
-        combined_names[ii + n_skip]  # Skip over the ones we left as the start
-    end
-    return compile_time_return_hack(ret)
+# dropped_dims must be a Tuple type where the values are Int literal for the dimensions being dropped
+@generated function _remaining_dimnames_after_dropping(dimnames::NTuple{N,Symbol}, dropped_dims::Type) where N
+    dropped_dims_vals = dropped_dims.parameters[1].parameters
+    keep_names = [:(getfield(dimnames, $ii)) for ii in 1:N if ii ∉ dropped_dims_vals]
+    return Expr(:call, :compile_time_return_hack, Expr(:tuple, keep_names...))
```

Comment:
I do not understand the evil/magic that is this function

perhaps you can say what advantage there is to writing this as an `@generated` function?

but i'm maybe not placed to review here

Reply:
Wring this as a generated function like this was the only way I could avoid allocations.
The particular avantage is that we basically use the `dropped_dims` type 
to exactly specify, at compile time,
the tuple we are returning,
in terms of which elements of the `dimnames` it contains
and then the `compile_time_return_hack` causes those to constant propegate
so this just all happens completely atc compiletime

Reply:
I am skeptical that this is the only way. I am going to try some things.

Reply:
Alright I couldn't get better things but we should have an issue to make this better